tests/conftest.py

```diff
@@ -154,18 +154,53 @@ def storageclass_factory_class(
 @pytest.fixture(scope='function')
 def storageclass_factory(
     request,
-    ceph_pool_factory,
-    secret_factory
 ):
     return storageclass_factory_fixture(
         request,
-        ceph_pool_factory,
-        secret_factory
     )
 
 
 def storageclass_factory_fixture(
     request,
+):
+    """
+    Return Default StorageClass for RBD or CephFS based on Interface
```

Comment:
I think this implementation adds unneeded complexity.
If you'll create a function (not fixture) to return the default (interface or storage class), you can keep using a single fixture for all with additional parameter (e.g. 'use_default=True'). 
With that, you will need to maintain a single factory and it makes it easier to understand

Reply:
My original plan was to just change the underlying layer to return defaults instead of create to see how it works without trying to change too many tests at once, so I kept the same old name. I am still testing as few tests are failing so I will am going to revisit the name and see what makes sense here after debugging those failures.